modules/javafx.controls/src/test/java/test/javafx/scene/control/ToolbarTest.java

```diff
@@ -221,4 +234,78 @@
         tk.firePulse();
         assertTrue(btn5.isFocused());
     }
+
+    @Test public void overflowShownInHorizontalAfterChildrenReziseTest() {
+        initializeToolBar();
+
+        toolBar.setOrientation(Orientation.HORIZONTAL);
+
+        testOverflowVisibility();
+    }
+
+    @Test public void overflowShownInVerticalAfterChildrenReziseTest() {
+        initializeToolBar();
+
+        toolBar.setOrientation(Orientation.VERTICAL);
+
+        testOverflowVisibility();
+    }
+
+    private void initializeToolBar() {
+        root = new StackPane(toolBar);
+        root.setPrefSize(400, 400);
+
+        scene = new Scene(root);
+
+        stage = new Stage();
+        stage.setScene(scene);
+
+        toolBar.getItems().addAll(node1, node2);
+        setFixSize(toolBar, TOOLBAR_SIZE);
+    }
+
+    private void testOverflowVisibility() {
+        setFixSize(node1, ORIGINAL_CHILDREN_SIZE);
+        setFixSize(node2, ORIGINAL_CHILDREN_SIZE);
+
+        assertOverflowNotShown();
+
+        setFixSize(node1, EXPANDED_CHILDREN_SIZE);
+
+        assertOverflowShown();
+
+        setFixSize(node1, ORIGINAL_CHILDREN_SIZE);
+
+        assertOverflowNotShown();
+    }
+
+    private void setFixSize(Node node, double size) {
+        if(node instanceof Region) {
+            ((Region) node).setMinSize(Pane.USE_PREF_SIZE, Pane.USE_PREF_SIZE);
+            ((Region) node).setMaxSize(Pane.USE_PREF_SIZE, Pane.USE_PREF_SIZE);
+            ((Region) node).setPrefSize(size, size);
+        } else if(node instanceof Rectangle) {
+            ((Rectangle) node).setHeight(size);
+            ((Rectangle) node).setWidth(size);
+        }
+
+        root.applyCss();
+        root.autosize();
+        root.layout();
+    }
+
+    private void assertOverflowNotShown() {
+        Pane pane = (Pane) toolBar.queryAccessibleAttribute(AccessibleAttribute.OVERFLOW_BUTTON);
+        if(pane!=null) {
```

Comment:
There should be a space between if and (.
Also around comparison operator

Correct version would be : if (pane != null)

Reply:
I think, you can use assertNotNull(pane); instead of if condition. This is already done in assertOverflowShown() method.

Reply:
You are right. overflowMenu is always initialized, and never should be null. I will leave it in two asserts and if it fails will be easier to find why.